projects/timer_challenge/src/main.c

```diff
@@ -0,0 +1,37 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+#define HALF_SEC 500
```

Comment:
I'm not sure I like this naming. Though it is clear what this is, someone reading this might get confused since they see a 500 beside HALF_SEC

Normally what we would do is name it something like DELAY_TIME_MS to indicate the units of the value and also to describe what this is used for